user/views.py

```diff
@@ -0,0 +1,41 @@
+import json
+import bcrypt
+import jwt
+
+from django.http    import JsonResponse
+from django.views   import View
+
+from wikea.settings import SECRET_KEY
+from my_settings    import ALGORITHM
+from user.models    import User
+
+class SignupView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            first_name      = data['first_name']
+            last_name       = data['last_name']
+            email           = data['email']
+            birthday        = data['birthday']
+            phone_number    = data['phone_number']
+            hashed_password = bcrypt.hashpw(data['password'].encode('utf-8'),bcrypt.gensalt()).decode('utf-8')
+            
+
+            if User.objects.filter(first_name = first_name, last_name = last_name).exists():
+                return JsonResponse({'message':'DUPLICATE NAME'}, status=409)
+            if User.objects.filter(email = email).exists():
+                return JsonResponse({'message':'DUPLICATE EMAIL'}, status=409)
+            if User.objects.filter(phone_number = phone_number).exists():
```

Comment:
```suggestion
            if User.objects.filter(first_name = first_name, last_name = last_name).exists():
                return JsonResponse({'message':'DUPLICATE NAME'}, status=409)
                
            if User.objects.filter(email = email).exists():
                return JsonResponse({'message':'DUPLICATE EMAIL'}, status=409)
                
            if User.objects.filter(phone_number = phone_number).exists():
```
- 가독성을 위해 if 절 간에 간격이 있으면 좋을 것 같습니다.
- 회원 가입을 진행할 때 유효성 검사를 진행하지 않는데, 프론트에서 한다고 해도 백엔드 입장에서는 프론트에서 들어오는 값들에 대해 100% 보장하기 어렵기 때문에 적절한 유효성 검사를 추가해 주는 것이 좋습니다.
- 유효성 검사를 추가하고 유효성 검사와 비밀번호 암호화의 순서에 대해 생각해 보세요.
- ststus code 409가 언제 발생하는지 코멘트로 남겨주세요.

Reply:
status 409는 request와 서버의 현재 상태와 충돌(conflict)했음을 알려줍니다.
회원 가입 시 입력한 이메일이 이미 가입돼 있는 이메일과 같다면 충돌했음을 알려주는 409를 쓰는게 좋을 것 같아서 사용했습니다.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409
이 글을 참고해 보면 보통 파일 업로드에서 버전 충돌 같은 상황에서 많이 사용되는 것을 볼 수 있습니다. 따라서 이 상황에서는 400이 더 낫지 않을까 생각됩니다.

Reply:
읽어보니깐 409는 보통 PUT 요청에 대한 응답으로 많이 사용하네요.. 400으로 수정하겠습니다.